source/slang/slang-check-decl.cpp

```diff
@@ -1948,8 +1948,57 @@ namespace Slang
         checkVisibility(classDecl);
     }
 
+    static InvokeExpr* _getDefaultInitOfVar(SemanticsVisitor* visitor, VarDeclBase* varDecl)
+    {
+        if (!varDecl->type || !varDecl->type.type)
+            return nullptr;
+
+        ConstructorDecl* defaultCtor = nullptr;
+        auto declRefType = as<DeclRefType>(varDecl->type.type);
+        if (declRefType)
+        {
+            if (auto structDecl = as<StructDecl>(declRefType->getDeclRef().getDecl()))
+            {
+                defaultCtor = _getDefaultCtor(structDecl);
+            }
+        }
+
+        if (defaultCtor)
+        {
+            auto* invoke = visitor->getASTBuilder()->create<InvokeExpr>();
+            auto member = visitor->getASTBuilder()->getMemberDeclRef(declRefType->getDeclRef(), defaultCtor);
+            invoke->functionExpr = visitor->ConstructDeclRefExpr(member, NULL, defaultCtor->loc, nullptr);
+            return invoke;
+        }
+        else
+        {
+            auto* defaultCall = visitor->getASTBuilder()->create<VarExpr>();
```

Comment:
This will work, but it will be faster to make `__default` a syntax instead of a function call right now, so we can do `__default(T)` that maps to `DefaultValExpr`, which lowers to the DefaultConstruct opcode. This saves us from running lookup and overload resolution logic for every variable initialization.

Reply:
note: added `DefaultConstructExpr` (to follow naming of `kIROp_DefaultConstruct`), I did not change the `__intrinsic_op` out (in slang-parser.cpp) since I was unclear on if we are to swap out this use of `__default` as well.

Reply:
Sure, that is fine.